src/name_bazaar/ui/utils.cljs

```diff
@@ -152,14 +152,41 @@
       (str name ".eth"))))
 
 (defn try-resolving-address [db addr]
-  (if-not (web3/address? addr)
+  (when-not (web3/address? addr)
     (let [addr-patched (tldize addr)]
-      (if-let [resolved (some (fn [r]
+      (when-let [resolved (some (fn [r]
                                 (get-in r [:public-resolver/records
                                            (namehash addr-patched)
                                            :public-resolver.record/addr]))
                               (vals (:public-resolvers db)))]
-        resolved
-        "0x"))
-    addr))
-
+        resolved))))
+
+(defn try-reverse-resolving-address [db addr]
+  (when (web3/address? addr)
+    (when-let [resolved (some (fn [r]
+                              (get-in r [:public-resolver/reverse-records
```

Comment:
Please incude spec for this into `name-bazaar.ui.spec`, also set it into db.cljs as `:public-resolver/reverse-records {}`